docs/reference/run-elasticsearch-locally.asciidoc

```diff
@@ -9,163 +15,75 @@
 
-[source,sh,subs="attributes"]
-----
-# configure the Kibana password in the ES container
-curl -u elastic:$ELASTIC_PASSWORD \
-  -X POST \
-  http://localhost:9200/_security/user/kibana_system/_password \
-  -d '{"password":"'"$KIBANA_PASSWORD"'"}' \
-  -H 'Content-Type: application/json'
+[source,sh]
+----     
+source .env
+curl $ES_LOCAL_URL -H "Authorization: ApiKey ${ES_LOCAL_API_KEY}"
 ----
 // NOTCONSOLE
 
-Start the {kib} container with the following command:
-
-ifeval::["{release-state}"=="unreleased"]
-WARNING: Version {version} has not yet been released.
-No Docker image is currently available for {es} {version}.
-endif::[]
-
-[source,sh,subs="attributes"]
-----
-docker run -p 127.0.0.1:5601:5601 -d --name kibana --network elastic-net \
-  -e ELASTICSEARCH_URL=http://elasticsearch:9200 \
-  -e ELASTICSEARCH_HOSTS=http://elasticsearch:9200 \
-  -e ELASTICSEARCH_USERNAME=kibana_system \
-  -e ELASTICSEARCH_PASSWORD=$KIBANA_PASSWORD \
-  -e "xpack.security.enabled=false" \
-  -e "xpack.license.self_generated.type=trial" \
-  {kib-docker-image}
-----
-
-When you access {kib}, use `elastic` as the username and the password you set earlier for the `ELASTIC_PASSWORD` environment variable.
-
-[NOTE]
+[CAUTION]
 ====
-The service is started with a trial license. The trial license enables all features of Elasticsearch for a trial period of 30 days. After the trial period expires, the license is downgraded to a basic license, which is free forever.
+This setup is for local testing only. HTTPS is disabled, and Basic authentication is used for {es}. For security, {es} and {kib} are accessible only via `localhost`.
 ====
 
 [discrete]
-[[local-dev-connecting-clients]]
-=== Connect to {es} with language clients
-
-To connect to the {es} cluster from a language client, you can use basic authentication with the `elastic` username and the password you set in the environment variable.
-
-.*Expand* for details
-[%collapsible]
-==============
-
-You'll use the following connection details:
-
-* **{es} endpoint**: `http://localhost:9200`
-* **Username**: `elastic`
-* **Password**: `$ELASTIC_PASSWORD` (Value you set in the environment variable)
-
-For example, to connect with the Python `elasticsearch` client:
-
-[source,python]
-----
-import os
-from elasticsearch import Elasticsearch
+[[local-dev-authentication]]
+=== Authentication
 
-username = 'elastic'
-password = os.getenv('ELASTIC_PASSWORD') # Value you set in the environment variable
+The script generates a random password for the `elastic` user, which is displayed at the end of the installation and stored in the `.env` file.
 
-client = Elasticsearch(
-    "http://localhost:9200",
-    basic_auth=(username, password)
-)
+An API key for {es} is generated and stored in the `.env` file as `ES_LOCAL_API_KEY`.
+Use this key to connect to {es} with a https://www.elastic.co/guide/en/elasticsearch/client/index.html[programming language client] or the <<rest-apis,REST API>>.
```

Comment:
I prefer the original placement of this in the readme - when I originally saw the "check the connection" step I didn't understand what it was doing